### ISSUE TITLE
Block PR Merging on Build Failure

### ISSUE THREAD

**Issue by MotoLegacy** [lavenderdotpet/LibreQuake#189]
We need to mandate that builds pass in order for a Pull Request to be merged, as we have continued merging and working on unrelated content while our builds are failing.

GitHub has a setting to absolutely block the merge button for PRs. We need to enable this and put everything else on hold until building is all green. This should he high priority.

**Comment by lavenderdotpet** [comment]
this issue popped up after the check was added

**Comment by MotoLegacy** [comment]
> this issue popped up after the check was added

Yes, it exposed several issues our maps have that were buried in logging. Now we have the resources to identify and resolve

**Comment by Zungrysoft** [comment]
At first, I thought this was already the case. I support adding this.

**Comment by pnahratow** [comment]
<img width="1761" height="1081" alt="Image" src="https://github.com/user-attachments/assets/f5279bb6-1ef8-4c57-b4a9-bbd9c29cb384" />

<img width="1180" height="806" alt="Image" src="https://github.com/user-attachments/assets/1b34bb97-d161-4f3d-9d68-5aad2c218b24" />

<img width="929" height="1154" alt="Image" src="https://github.com/user-attachments/assets/76c7a0f6-c3cd-43fc-b102-368012ec0452" />

**Comment by lavenderdotpet** [comment]
<img width="793" height="149" alt="Image" src="https://github.com/user-attachments/assets/12393ccc-d63b-49dd-9001-ce3ceb71773d" />

**Comment by pnahratow** [comment]
Apparently this a super common problem with no clean solution provided by github yet. Please disable the rule for now. I will try to come up with a solution. 

The documentation I could find had solutions like

- Do the path filtering in the script and just return success if the check is not needed
- Have workflow file with the opposite path filtering and all the jobs with the same name and have them return success

This sounds super dumb.

I'll see if I can test if there were no failed jobs and have that as a mandatory job on which the rule depends. I'll get back to it.
